tools/autograd/templates/Functions.cpp

```diff
@@ -364,17 +364,23 @@ Tensor unsqueeze_to(const Tensor & self, int64_t dim, IntList sizes) {
   return self;
 }
 
-std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<int64_t> &sizes, int64_t dim) {
+std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, int64_t dim) {
+  if (sizes.size() > 0) {
+    // cat wraps dim to the first tensor's shape 
+    dim = at::maybe_wrap_dim(dim, sizes[0].size());
```

Comment:
can we move this logic into wrap_dim_utils?  In another PR you change how TensorList wrap_dim works, but it will be hard to remember to update this.

Reply:
My intention was to delete this piece in #5837 after this PR gets merged in because it won't be necessary anymore. I can add it as a new function in wrap_dim_utils but it'll get deleted by #5837 anyways.

Reply:
I'm not sure I buy https://github.com/pytorch/pytorch/pull/5837 yet, can we make this as standalone as possible?

Reply:
Sounds good!